uvicorn/main.py

```diff
@@ -359,7 +359,11 @@ def run(self, sockets=None, shutdown_servers=True):
 
         elif config.uds is not None:
             # Create a socket using UNIX domain socket.
+            uds_perms = 0o666
+            if os.path.exists(config.uds):
+                uds_perms = os.stat(config.uds).st_mode
```

Comment:
Rather thank do this, it'd make more sense simply to *not* perform `os.chmod` if `os.path.exists(...)` returned true.

Eg. Use `should_chmod = not os.path.exists(config.uds)` here.

(Eg. perhaps the user account might not have permissions to chmod it for whatever reason)

Reply:
Thanks for your time in getting this ironed out!
It'll end up being a relatively small PR for quite a lot of chat, but it's worth getting it right.

Reply:
create_unix_server resets permissions (0755), so my code returns it back which it had before

Reply:
Ah gotcha